src/payment/strategies/chasepay/chasepay-payment-strategy.ts

```diff
@@ -0,0 +1,224 @@
+        const { methodId } = options;
+
+        if (!options.chasepay) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.chasepayCheckoutOptions" argument is not provided.');
+        }
+
+        const {
+            onPaymentSelect = () => {},
+            logoContainer,
+            chasePayActionButton,
+        } = options.chasepay;
+
+        this._logoContainer = logoContainer;
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId))
+            .then(state => {
+                const storeConfig = state.config.getStoreConfig();
+
+                this._paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+                if (!this._paymentMethod || !this._paymentMethod.initializationData) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                if (!storeConfig) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+                }
+
+                if (options.gatewayId) {
+                    this._wepayRiskClient.initialize();
+                }
+
+                if (document.getElementById(chasePayActionButton || '')) {
+                    const chasePayActionButtonListener = document.body.querySelector(`#${chasePayActionButton}`) as HTMLElement;
+                    chasePayActionButtonListener.addEventListener('click', () =>
+                        this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(options.methodId))
+                            .then(() => this._displayChasepayLightbox(this._paymentMethod)));
+                }
+
+                return this._setChasePayConfiguration(
+                    methodId,
+                    storeConfig.storeProfile.storeLanguage,
+                    this._paymentMethod.config.testMode,
+                    onPaymentSelect
+                );
+            })
+        .then(() => super.initialize(options));
+    }
+
+    execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = orderRequest;
+
+        if (!payment || !options) {
+            throw new InvalidArgumentError('Unable to proceed because the customer has not signed into their ChasePay wallet account or selected a payment option.');
+        }
+
+        if (!this._paymentMethod || !this._paymentMethod.initializationData) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        const { initializationData } = this._paymentMethod;
+        const riskToken = (payment.gatewayId === 'wepay') ? this._wepayRiskClient.getRiskToken() : undefined;
+        const paymentData = this._mapCardData(initializationData, riskToken);
+
+        return this._store.dispatch(this._orderActionCreator.submitOrder(order, options))
```

Comment:
Ok, do you know that when the client does the following:

```ts
checkoutService.submitOrder({ methodId: 'chasepay' });
```

We locate the `ChasePayStrategy` and call its `execute` method. What do people normally expect `submitOrder` do? They expect it to submit the order. But we can't submit the order without payment details. That's why we prompt the customer to enter their payment details. We do that for PayPal and other hosted payment methods. That's why I suggest we should do the same thing unless there's a special reason why ChasePay has to be different. If it's 100% not possible for you to make the behaviour consistent with all other hosted payment methods, you need to throw an error here if the customer hasn't select a credit card from their wallet, and make `walletButton` a compulsory option. Personally, I wouldn't recommend that option unless there's no other workaround.

Reply:
Can we talk about this one? It seems that I wasn't able to replicate the behavior that you asked for.

Reply:
I think that creating this behavior is out of the scope of this user story since the original implementation on `ng-checkout` does not handle this behavior.
So, to handle this event in case that the user calls `checkoutService.submitOrder({ methodId: 'chasepay' });` we just trigger the error `throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);`

In any case I'll check with @cbandam and Kyle to create a new task for this other implementation. 

What do you think about this?

Reply:
I understand that you weren't initially aware of this behaviour, therefore it wasn't scoped in your ticket. But it needs to be done first before you can merge this PR. Otherwise, we will be making a breaking change once the second part is done. Alternatively, we could merge this PR in the dark - meaning not registering this payment strategy so it has no effect on anything.

Anyway, I can quickly check out your branch to see how complicated it is to get it working. If it's quite complicated, I'd go with the second approach. Otherwise, I'd just do the work as part of this PR.

Reply:
I think we found a better way to handle this behavior. Once I have it ready I'll update the PR so you can review it.

Reply:
@davidchin Can you review the latest code to check the implementation that we make to handle this?